ERROR while compiling ocaml-secondary-compiler.4.08.1

Hi, I am currently trying to install `ocaml-secondary-compiler.4.08.1`, and opam gave an error while building with the following error message. 

```
[ERROR] The compilation of ocaml-secondary-compiler failed at "/Users/irene/.opam/opam-init/hooks/sandbox.sh build make -j3 world.opt".

#=== ERROR while compiling ocaml-secondary-compiler.4.08.1 ====================#
# context     2.0.5 | macos/x86_64 | ocaml-variants.4.02.2+multicore | https://opam.ocaml.org#8e32fa53
# path        ~/.opam/4.02.2+multicore/.opam-switch/build/ocaml-secondary-compiler.4.08.1
# command     ~/.opam/opam-init/hooks/sandbox.sh build make -j3 world.opt
# exit-code   2
# env-file    ~/.opam/log/ocaml-secondary-compiler-77298-1114b2.env
# output-file ~/.opam/log/ocaml-secondary-compiler-77298-1114b2.out
### output ###
# [...]
# ../../boot/ocamlrun ../../ocamlc -nostdlib -I ../../stdlib -c -absname -w +a-4-9-41-42-44-45-48 -warn-error A -bin-annot -g -safe-string -strict-sequence -strict-formats  graphicsX11.mli
# ../../boot/ocamlrun ../../ocamlc -nostdlib -I ../../stdlib -c -absname -w +a-4-9-41-42-44-45-48 -warn-error A -bin-annot -g -safe-string -strict-sequence -strict-formats  graphics.ml
# ../../boot/ocamlrun ../../ocamlc -nostdlib -I ../../stdlib -c -absname -w +a-4-9-41-42-44-45-48 -warn-error A -bin-annot -g -safe-string -strict-sequence -strict-formats  graphicsX11.ml
# ../../boot/ocamlrun ../../tools/ocamlmklib -oc graphics open.o draw.o fill.o color.o text.o image.o make_img.o dump_img.o point_col.o sound.o events.o subwindow.o -ldopt "-L/usr/local/lib -lX11"
# ld: library not found for -lX11
# clang: error: linker command failed with exit code 1 (use -v to see invocation)
# make[4]: *** [libgraphics.a] Error 2
# make[4]: *** Waiting for unfinished jobs....
# make[3]: *** [all] Error 2
# make[2]: *** [otherlibraries] Error 2
# make[1]: *** [opt.opt] Error 2
# make: *** [world.opt] Error 2
```

```
<><> ocaml-secondary-compiler.4.08.1 troubleshooting ><><><><><><><><><><><>  🐫
=> A failure in the middle of the build may be caused by build parallelism
      (enabled by default).
Please file a bug report at https://github.com/ocaml/ocaml/issues
```

I tried installing `reagents` again using `--jobs=1` to force a sequential build, and it still gave me the same error message. I'm currently on `ocaml` version 4.02.2 with support for Multicore OCaml (`ocaml-variants` 4.02.2+multicore).

--- a/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1-1/files/0001-Don-t-build-manpages-for-stdlib-docs.patch
+++ b/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1-1/files/0001-Don-t-build-manpages-for-stdlib-docs.patch
@@ -1,0 +1,24 @@
+From 0cf3c6ad7ce2a2b2806faceccfb0a9321da5e22a Mon Sep 17 00:00:00 2001
+From: David Allsopp <david.allsopp@metastack.com>
+Date: Fri, 26 Jul 2019 12:12:19 +0100
+Subject: [PATCH] Don't build manpages for stdlib docs
+---
+ ocamldoc/Makefile            | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/ocamldoc/Makefile b/ocamldoc/Makefile
+index b109815071..e31e441f61 100644
+--- a/ocamldoc/Makefile
++++ b/ocamldoc/Makefile
+@@ -170,7 +170,7 @@ LIBCMIFILES = $(LIBCMOFILES:.cmo=.cmi)
+ 
+ 
+ .PHONY: all
+-all: lib exe generators manpages
++all: lib exe generators
+ 
+ manpages: generators
+ 
+-- 
+2.20.1
+

--- a/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1-1/files/0001-Fix-failure-to-install-tools-links.patch
+++ b/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1-1/files/0001-Fix-failure-to-install-tools-links.patch
@@ -1,0 +1,26 @@
+From 705739fa54260b7a0e6cbba0b5a99e52c79f9c09 Mon Sep 17 00:00:00 2001
+From: David Allsopp <david.allsopp@metastack.com>
+Date: Tue, 6 Aug 2019 09:23:06 +0100
+Subject: [PATCH] Fix failure to install tools links
+
+In --disable-installing-bytecode-programs mode, the .opt version of the
+tools is installed, but the symlink for the tool itself is not created.
+---
+ tools/Makefile | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/tools/Makefile b/tools/Makefile
+index 530dd37f34..1b3014a3ab 100644
+--- a/tools/Makefile
++++ b/tools/Makefile
+@@ -197,6 +197,7 @@ else
+ 	do \
+ 	  if test -f "$$i".opt; then \
+ 	    $(INSTALL_PROG) "$$i.opt" "$(INSTALL_BINDIR)/$$i.opt$(EXE)"; \
++	    (cd "$(INSTALL_BINDIR)/" && $(LN) "$$i.opt$(EXE)" "$$i$(EXE)"); \
+ 	  fi; \
+ 	done
+ endif
+-- 
+2.20.1
+

--- a/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1-1/opam
+++ b/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1-1/opam
@@ -15,6 +15,7 @@ build: [
       "--disable-installing-bytecode-programs"
       "--disable-debug-runtime"
       "--disable-instrumented-runtime"
+      "--disable-graph-lib"
       "CC=cc" {os = "openbsd" | os = "freebsd" | os = "macos"}
       "ASPP=cc -c" {os = "openbsd" | os = "freebsd" | os = "macos"}
   ]

--- a/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1-1/opam
+++ b/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1-1/opam
@@ -5,7 +5,7 @@ authors: "Xavier Leroy and many contributors"
 homepage: "https://ocaml.org"
 bug-reports: "https://github.com/ocaml/ocaml/issues"
 dev-repo: "git://github.com/ocaml/ocaml"
-conflicts: "ocaml" {>= "4.08.0" & < "4.09~"}
+depends: "ocaml" {< "4.08.0" | >= "4.09~"}
 build: [
   [
     "./configure"

--- a/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1/opam
+++ b/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1/opam
@@ -15,6 +15,7 @@ build: [
       "--disable-installing-bytecode-programs"
       "--disable-debug-runtime"
       "--disable-instrumented-runtime"
+      "--disable-graph-lib"
       "CC=cc" {os = "openbsd" | os = "freebsd" | os = "macos"}
       "ASPP=cc -c" {os = "openbsd" | os = "freebsd" | os = "macos"}
   ]

--- a/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1/opam
+++ b/packages/ocaml-secondary-compiler/ocaml-secondary-compiler.4.08.1/opam
@@ -5,7 +5,7 @@ authors: "Xavier Leroy and many contributors"
 homepage: "https://ocaml.org"
 bug-reports: "https://github.com/ocaml/ocaml/issues"
 dev-repo: "git://github.com/ocaml/ocaml"
-conflicts: "ocaml" {>= "4.08.0" & < "4.09~"}
+depends: "ocaml" {< "4.08.0" | >= "4.09~"}
 build: [
   [
     "./configure"

